recipes/flatbuffers/all/conanfile.py

```diff
@@ -38,32 +38,43 @@ def config_options(self):
+        cmake.definitions["FLATBUFFERS_BUILD_FLATLIB"] = not self.options.shared and not self.options.header_only
         cmake.definitions["FLATBUFFERS_BUILD_FLATC"] = False
         cmake.definitions["FLATBUFFERS_BUILD_FLATHASH"] = False
         cmake.configure(build_folder=self._build_subfolder)
         return cmake
 
     def build(self):
-        cmake = self._configure_cmake()
-        cmake.build()
+        if not self.options.header_only:
+            cmake = self._configure_cmake()
+            cmake.build()
 
     def package(self):
         self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
-        cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-        if self.settings.os == "Windows" and self.options.shared:
-            if self.settings.compiler == "Visual Studio":
-                tools.mkdir(os.path.join(self.package_folder, "bin"))
-                shutil.move(os.path.join(self.package_folder, "lib", "%s.dll" % self.name),
-                            os.path.join(self.package_folder, "bin", "%s.dll" % self.name))
-            elif self.settings.compiler == "gcc":
-                shutil.move(os.path.join(self.package_folder, "lib", "lib%s.dll" % self.name),
-                            os.path.join(self.package_folder, "bin", "lib%s.dll" % self.name))
+        if self.options.header_only:
+            header_dir = os.path.join(self._source_subfolder, "include", "flatbuffers")
+            dst_dir = os.path.join("include", "flatbuffers")
+            self.copy("*.h", dst=dst_dir, src=header_dir)
+        else:
+            cmake = self._configure_cmake()
+            cmake.install()
+            tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+            if self.settings.os == "Windows" and self.options.shared:
+                if self.settings.compiler == "Visual Studio":
+                    tools.mkdir(os.path.join(self.package_folder, "bin"))
+                    shutil.move(os.path.join(self.package_folder, "lib", "%s.dll" % self.name),
+                                os.path.join(self.package_folder, "bin", "%s.dll" % self.name))
+                elif self.settings.compiler == "gcc":
+                    shutil.move(os.path.join(self.package_folder, "lib", "lib%s.dll" % self.name),
+                                os.path.join(self.package_folder, "bin", "lib%s.dll" % self.name))
+
+    def package_id(self):
+        if self.options.header_only:
+            self.info.header_only()
 
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
-        if self.settings.os == "Linux":
-            self.cpp_info.libs.append("m")
+        if not self.options.header_only:
+            self.cpp_info.libs = tools.collect_libs(self)
+            if self.settings.os == "Linux":
+                self.cpp_info.libs.append("m")
```

Comment:
For header-only we could add `self.cpp_info.defines = ["FLATBUFFERS_HEADER_ONLY"]` and it will simplify the PR a little bit, right?

Reply:
You mean for use in the test_package? I'm a little hesitant as the original flatbuffers lib does not do that. This means if users start using this define it is not as simple as before to switch between using conan or something else.

Or are there more advantages to adding this define besides the test_package?

Reply:
I prefer keeping it in test package. Adding new defines can cause confusion when building, and a minimal risk for conflicts in future releases.